src/Cell.cpp

```diff
@@ -250,6 +287,56 @@ std::map<int, surface_halfspace> Cell::getSurfaces() const {
 
 
 /**
+ * @brief Returns the std::map of Cell IDs and Cell pointers within any
+ *        nested Universes filling this Cell.
+ * @return std::map of Cell IDs and pointers
+ */
+std::map<int, Cell*> Cell::getAllCells() {
+
+  std::map<int, Cell*> cells;
+
+  if (_cell_type == FILL && _fill != NULL) {
+    std::map<int, Cell*> nested_cells;
+    Universe* univ_fill = static_cast<Universe*>(_fill);
+
+    if (univ_fill->getType() == SIMPLE)
+      nested_cells = univ_fill->getAllCells();
+    else
+      nested_cells = static_cast<Lattice*>(univ_fill)->getAllCells();
+
+    cells.insert(nested_cells.begin(), nested_cells.end());
+  }
+
+  return cells;
+}
+
+
+/**
+ * @brief Returns the std::map of all nested Universe IDs and Universe pointers
+          filling this Cell.
+ * @return std::map of Universe IDs and pointers
+ */
+std::map<int, Universe*> Cell::getAllUniverses() {
+
+  std::map<int, Universe*> universes;
+
+  if (_cell_type == FILL && _fill != NULL) {
+    Universe* univ_fill = static_cast<Universe*>(_fill);
+    universes[univ_fill->getId()] = univ_fill;
+
+    std::map<int, Universe*> nested_universes;
+    if (univ_fill->getType() == SIMPLE)
+      nested_universes = static_cast<Universe*>(_fill)->getAllUniverses();
+    else
+      nested_universes = static_cast<Lattice*>(_fill)->getAllUniverses();
+    universes.insert(nested_universes.begin(), nested_universes.end());
+  }
+
+  return universes;
```

Comment:
Ok, so is there anything I need to do to appease you with these empty maps, or do they make sense?